### ISSUE TITLE
No CRC or fall back

### ISSUE THREAD

**Issue by daverathbone** [chrisjoyce911/esp32FOTA#71]
The code does not perform any CRC check on the bin file downloaded from the server?
The rest of the error handling is just serial print statements as place holders in the library.
Ideally these  errors should be on a callback  so that the users code can decide what to do with it.

When the code is transferred to flash if the CRC fails it should roll back to the last known working version. This will keep repeating until it gets the update.

The Espressif ESP-IDF examples has  HTTP and HTTPS that does this, however... I was looking for an Arduino one like yours...as I can have the benefit of 100's of other Arduino libraries for ESP32 .

**Comment by thorrak** [comment]
Can you link one of those examples you are referencing?

**Comment by daverathbone** [comment]
Within Espressif's examples and API's see [https://github.com/espressif/esp-idf/blob/master/examples/system/ota/simple_ota_example/main/simple_ota_example.c](url) 

I first tested with your code only as I need other Arduino framework items for  ease of development within Platformio.
(Thank you)

As I outlined it would be handy if the Serial print or log errors came back as a call back, to the users main loop.
Allowing for user to error handling stuff, rather than modifying your library. 

I also found that I did not need certificate based handshakes for my HTTPS server as its encrypted any way. (_HTTPS ssl_)   
I just send  a username and password, in the HTTPs  'get' header which an Apache server uses to allow the JSON file access.
This is well secure enough, as the JSON reply has the real url for the firmware bin in it, which is well hidden from any 'man  in the middle' risks or server sniffing Bots.  

The standard trick is to send servers massive urls in order to break buffers., home grown node.js servers will fall over  other servers are known to spill the beans.  In the main it can just give you denial of service risks.

I've modified your library to use a local certificate or not on  a setup call.
Just remember that many self signed SSL certs,  will run out quicker and your embedded device will be lost at sea.

Note: The Arduino library to your code downloads from Platformio and gets the older version that does not have http or https switch . Which is up-to-date on this GIT repo.
